docs/data_quality.md

```diff
@@ -0,0 +1,111 @@
+
+## Examples
+
+* [GO-2024-2687](https://api.osv.dev/v1/vulns/GO-2024-2687)
+  * Has `introduced` and `fixed` versions
+  * Has an alias to a CVE record ID
+  * Has a purl
+* [OSV-2024-98](https://api.osv.dev/v1/vulns/OSV-2024-98)
+  * Has `introduced` and `fixed` commits
+    * commits exist in repo
+* [DSA-5678-1](https://api.osv.dev/v1/vulns/DSA-5678-1)
+  * Has `introduced` and `fixed` versions
+  * Has multiple `related` CVE record IDs
+
+## Appendix A: OSV Schema validation
+
+(As at version 1.6.3, generated by Gemini)
+
+**Top-Level Information:**
+
+* **id:** A unique string identifier for the vulnerability.
+* **modified:** A timestamp (in a specific format) indicating when the vulnerability information was last updated.
+
+**Optional, but validated when present:**
+
+* **schema\_version:** A string specifying the version of the schema being used.
+* **published/withdrawn:** Timestamps for when the vulnerability was published or withdrawn.
+* **aliases/related:** Arrays of strings for alternate identifiers or related vulnerabilities.
+* **summary/details:** String descriptions of the vulnerability.
+* **severity:** An array of objects detailing the severity using different scoring systems (e.g., CVSS v2, v3, or v4), if available.
+* **affected:** An array of objects describing which packages are affected, including details like:
+  * **package:** The ecosystem (e.g., npm, PyPI), name, and Package URL (PURL) of the affected package.
+  * **severity:** Severity for the specific package (if different from the overall severity).
+  * **ranges:** Information on the affected version ranges, commit ranges, or ecosystem-specific identifiers.
+  * **versions:** A list of specific affected versions.
+  * **ecosystem\_specific/database\_specific:** Additional data specific to the package ecosystem or the vulnerability database.
+* **references:** An array of objects providing URLs to external resources about the vulnerability, categorized by type (e.g., advisory, article, discussion).
+* **credits:** An array of objects giving credit to individuals or organizations involved in discovering, reporting, or fixing the vulnerability.
+* **database\_specific:** A flexible object for any extra information specific to the database using this schema.
+
+**Additional Validation Rules:**
+
+* **timestamp:** A custom definition that ensures timestamps adhere to a specific date-time format (e.g., "2023-11-15T12:34:56Z").
+* **additionalProperties: false:** This prevents any extra properties from being added to the JSON object beyond those defined in the schema.
+* **Specific Requirements in `affected` Array:
+  * There are conditional validations based on the `type` of range, ensuring the correct properties are present (e.g., `repo` is required when `type` is `GIT`).
+  * A logical check ensures that if `last_affected` is specified in `events`, then `fixed` cannot be present in the same `events` array.
+
+**Overall:**
```

Comment:
nit: this part is probably redundant for this doc.

Reply:
Removed in 2e954725